lib/dradis/plugins/mapping_service.rb

```diff
@@ -0,0 +1,77 @@
+module Dradis
+  module Plugins
+    class MappingService
+      attr_accessor :component, :destination, :integration, :source
+
+      def initialize(args = {})
+        @component = args[:component]
+        @integration = args[:integration]
+        @destination = args[:destination]
+        @sample_dir = args[:sample_dir] || default_sample_dir
+      end
+
+      def apply_mapping(args = {})
+        @source = args[:source] || source
+        return unless validate_source
+        data = args[:data]
+        field_processor = integration::FieldProcessor.new(data: data)
+        mapping_fields = args[:mapping_fields] || get_mapping_fields
+
+        mapping_fields.map do |field|
+          field_name = field.try(:destination_field) || field[0]
+          field_content = process_content(
+            field.try(:content) || field[1],
+            field_processor
+          )
+
+          "#[#{field_name}]#\n#{field_content}"
+        end&.join("\n\n")
+      end
+
+      # This returns a sample of valid entry for the Mappings Manager
+      def sample
+        @sample ||= {}
+        if validate_source
+            @sample[source] ||= begin
+            sample_file = File.join(@sample_dir, "#{source}.sample")
+            File.read(sample_file)
```

Comment:
do we want to cache these on Redis? We use that in the Dashboard for What's New and the Integrations list.

```
Rails.cache.fetch("files_cache.#{Dradis::Pro::VERSION::STRING}.integrations")
```